asmcomp/amd64/arch.ml

```diff
@@ -134,7 +134,20 @@ let print_specific_operation printreg op ppf arg =
   | Izextend32 ->
       fprintf ppf "zextend32 %a" printreg arg.(0)
 
+(* Are we using the Windows 64-bit ABI? *)
+
 let win64 =
   match Config.system with
   | "win64" | "mingw64" | "cygwin" -> true
   | _                   -> false
+
+(* Specific operations that are pure *)
+
+let operation_is_pure = function
+  | Ilea _ | Ibswap _ | Isqrtf | Isextend32 | Izextend32 -> true
+  | Ifloatarithmem _ | Ifloatsqrtf _ -> true
```

Comment:
Note to reviewers: the "float arithmetic with one operand loaded from memory" operations were not considered pure in the previous implementation, but I am positive that they are!  Probably they were not considered pure before by analogy with the corresponding specific FP operations in the i386 port.  But in i386, FP operations have side effects on the x87 FP stack, while this is not the case in amd64 (SSE2 FP is used instead of x87 FP).